app/src/main/kotlin/com/wire/android/ui/home/userprofile/UserProfileScreen.kt

```diff
@@ -1,16 +1,343 @@
+import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
-import com.wire.android.ui.common.UnderConstructionScreen
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.wire.android.R
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.common.Icon
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.common.UserStatus
+import com.wire.android.ui.common.UserStatusDot
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryButton
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.selectableBackground
+import com.wire.android.ui.common.textfield.WirePrimaryButton
+import com.wire.android.ui.theme.wireTypography
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun UserProfileScreen() {
-    UnderConstructionScreen(screenName = "UserProfileScreen")
+fun UserProfileScreen(viewModel: UserProfileViewModel) {
+    val uiState by viewModel.state.collectAsState()
+
+    with(uiState) {
+        renderUserProfileScreen(state = this, viewModel = viewModel)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun renderUserProfileScreen(state: UserProfileState, viewModel: UserProfileViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
+            .background(MaterialTheme.colorScheme.background)
+    ) {
+
+        renderTopBar(viewModel)
+        renderHeader(state, viewModel)
+        renderStatusesRow(state)
+        renderOtherAccountsList(state, viewModel)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ColumnScope.renderTopBar(viewModel: UserProfileViewModel) {
+    CenterAlignedTopAppBar(
+        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.background,
+            titleContentColor = MaterialTheme.colorScheme.onBackground,
+            actionIconContentColor = MaterialTheme.colorScheme.onBackground,
+            navigationIconContentColor = MaterialTheme.colorScheme.onBackground
+        ),
+        navigationIcon = {
+            IconButton(onClick = { viewModel.close() }) {
+                Icon(
+                    imageVector = Icons.Filled.Close,
+                    contentDescription = stringResource(R.string.user_profile_close_description),
+                    tint = MaterialTheme.colorScheme.onBackground
+                )
+            }
+        },
+        title = {
+            Text(
+                text = stringResource(id = R.string.user_profile_tile),
+                style = MaterialTheme.wireTypography.title01,
+                color = MaterialTheme.colorScheme.onBackground,
+            )
+        },
+        actions = {
+            //TODO make it red, when such WireBtn ready
```

Comment:
Can't we already specify it via the `colors` parameter?

Reply:
I could, but `colors` is `interface WireButtonColors` so i'd need to provide all the colors, but just "textColor" for example. Also Michael mentioned he is going to add an additional btn state "Error", which will be red and looks just like I need here :)

Reply:
It's already in the `develop` - you just use `WireButtonState.Error` 😄

Reply:
More than reusing an error state button, I'd say it's better to add a `textColor` parameter in the constructor, as the error button might end up having some slightly different behavior in the future.

Reply:
I was thinking about that, but the name "error" is because of the semantic colors approach. We have a limiter number of button types and states based on the designs so we should use these predefined ones and don't mix or modify them freely.